app/common/lib/bookmarkUtil.js

```diff
@@ -119,9 +119,9 @@ const getToolbarBookmarks = (state) => {
   }
 
   return {
-    visibleBookmarks: noParentItems.take(i),
+    visibleBookmarks: noParentItems.take(i).map((item, index) => index).toList(),
```

Comment:
Instead of sending the whole object we are sending only keys. So we have list of keys and this way we are not triggering re-renders. We send this key to the child components which gets the whole object from the state.